homeassistant/components/flexit_bacnet/config_flow.py

```diff
@@ -0,0 +1,78 @@
+"""Config flow for Flexit Nordic (BACnet) integration."""
+from __future__ import annotations
+
+import asyncio.exceptions
+import logging
+from typing import Any
+
+from flexit_bacnet import FlexitBACnet
+from flexit_bacnet.bacnet import DecodingError
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.const import CONF_DEVICE_ID, CONF_IP_ADDRESS
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+DEFAULT_DEVICE_ID = 2
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_IP_ADDRESS): str,
+        vol.Required(CONF_DEVICE_ID, default=DEFAULT_DEVICE_ID): int,
+    }
+)
+
+
+async def validate_input(data: dict[str, Any]) -> str:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    device = FlexitBACnet(data[CONF_IP_ADDRESS], data[CONF_DEVICE_ID])
+
+    try:
+        await device.update()
+    except (asyncio.exceptions.TimeoutError, ConnectionError, DecodingError) as exc:
+        raise CannotConnect from exc
+
+    # Return info that you want to store in the config entry.
+    return str(device.serial_number)
```

Comment:
what is serial number in the backend? is it a str already or is it an int?
If it is a str you shouldn't be calling this

Reply:
Missed that one. It's removed now.